internal/agent.go

```diff
@@ -101,6 +106,18 @@ func (a *Agent) Start() error {
 		log.Info("heartbeat loop stopped.")
 	}(&wg)
 
+	if a.config.FactsEngineEnabled {
```

Comment:
This goroutines are somehow wired together? In terms of errors propagation? If yes we can think of using a errgroup with the same context, in order to coordinate the failure/context injection. The code should be the same, I think that we can get rid of the waitgroup also, the errgroup has this similar waitgroup implementation inside.

Reply:
Done!
Thank you for the tip